little-books/src/components/BlogDetail/BlogDetail.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useState } from "react";
+import { BlogType } from "../../types/blog.type";
+import Button from "../Button/Button";
+import "./BlogDetail.scss";
+import { useAppDispatch } from "../../hooks";
+import { updateBlog } from "../../redux/blogs/blogSlice";
+import FallbackImage from "../FallbackImage/FallbackImage";
+
+interface BlogDetailProps {
+	item: BlogType;
+}
+
+const BlogDetail: React.FC<BlogDetailProps> = ({ item }) => {
+	const dispatch = useAppDispatch();
+
+	const [isEdit, setIsEdit] = useState(false);
+	const [formObj, setFormObj] = useState<BlogType>({ ...item });
+
+	useEffect(() => {
+		setFormObj({ ...item }); // Update formObj when item changes
+	}, [item]);
+
+	const handleEditButton = () => {
+		setIsEdit(true);
+	};
+
+	const handleSaveButton = () => {
+		setIsEdit(false);
+		dispatch(updateBlog(formObj));
+	};
+
+	const handleCancelButton = () => {
+		setIsEdit(false);
+		setFormObj({ ...item });
+	};
+
+	const handleSelectChange = (value: string, name: string) => {
+		setFormObj((prev) => ({ ...prev, [name]: value }));
+	};
+
+	return (
+		<div className="blog-details-wrapper">
+			{isEdit ? (
+				<div className="blog-edit-container">
+					<div className="blog-detail-img">
+						<FallbackImage className="blog-img" 
+						fallbackUrl={"https://dummyimage.com/16:9x1080/';"} 
+						imgUrl={item?.photo} alt="Blog Image"						
+						/>
+					</div>
+					<form>
+						<div className="input-item">
+							<input
+								type="text"
+								name="title"
+								value={formObj.title}
+								onChange={(e) =>
+									handleSelectChange(
+										e.target.value,
+										e.target.name
+									)
+								}
+							/>
+						</div>
+						<textarea
+							className="blog-description"
+							name="details"
+							value={formObj.details}
+							onChange={(e) =>
+								handleSelectChange(
+									e.target.value,
+									e.target.name
+								)
+							}
+						></textarea>
```

Comment:
Have the same form for both editable state and showing details with setting the input as readOnly & hiding the outline & borders

Reply:
resolved